### PR TITLE
Always set password when exporting from DSM

### LINKED ISSUE #38: Support `--with-password`

### PR DESCRIPTION
**Pull request by zugzwang**
Fixes #38 but first merge #41

### CODE REVIEW

File: sq/tests/dsm/extract_dsm_import_gpg.sh [hunk 1]
```diff
@@ -124,9 +127,11 @@ $sq encrypt --signer-dsm-key="$bob_key_name" --recipient-cert "$alice_public" "$
...
 
 comm "decrypt with (i) local key (ii) SDKMS key (iii) gpg-imported key"
-$gpg --output="$decrypted_gpg" --decrypt "$encrypted"
-diff "$message" "$decrypted_gpg"
 $sq decrypt --signer-cert="$bob_public" --recipient-key="$alice_local_priv" "$encrypted" --output "$decrypted_local"
 diff "$message" "$decrypted_local"
 $sq decrypt --signer-cert="$bob_public" --dsm-key="$alice_key_name" "$encrypted" --output "$decrypted_remote"
 diff "$message" "$decrypted_remote"
+$gpg --passphrase="test_passphrase" --output="$decrypted_gpg" --decrypt "$encrypted"
+diff "$message" "$decrypted_local"
```
Comment (xinyufort) [inline sq/tests/dsm/extract_dsm_import_gpg.sh:135]: Is this supposed to be `$decrypted_gpg`? Or is `$decrypted_local` correct?

**Review by xinyufort — COMMENTED** [review]
I presume you have tested this out, right? Also see my comment re `$decrypted_gpg`

**Comment by xinyufort** [comment]
Also, if applicable, make sure to add RSA 2048 support to this PR too (perhaps merge PR #42 first)

File: sq/tests/dsm/extract_dsm_import_gpg.sh (hunk 1 above)
Comment (zugzwang) [reply sq/tests/dsm/extract_dsm_import_gpg.sh:135]: Good catch, it got reverted/unmodified in a rebase (see https://github.com/fortanix/sq-dsm/issues/40)

**Review by xinyufort — APPROVED** [review]

**Pull request merged**